lib/integrations/samlidp/gcpworkforce.go

```diff
@@ -0,0 +1,212 @@
+	newGCPWorkforceService := &GCPWorkforceService{
+		APIParams: samlidpconfig.GCPWorkforceAPIParams{
+			PoolName:           cfg.APIParams.PoolName,
+			PoolProviderName:   cfg.APIParams.PoolProviderName,
+			OrganizationID:     cfg.APIParams.OrganizationID,
+			SAMLIdPMetadataURL: cfg.APIParams.SAMLIdPMetadataURL,
+		},
+		HTTPClient: cfg.HTTPClient,
+	}
+
+	if err := newGCPWorkforceService.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return newGCPWorkforceService, nil
+}
+
+// CreateWorkforcePoolAndProvider creates GCP Workforce Identity Federation pool and pool provider
+// with the given GCPWorkforceAPIParams values.
+func (s *GCPWorkforceService) CreateWorkforcePoolAndProvider(ctx context.Context) error {
+	fmt.Println("\nConfiguring Workforce Identity Federation pool and SAML provider.")
+
+	iamService, err := iam.NewService(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	workforceService := iam.NewLocationsWorkforcePoolsService(iamService)
+
+	fmt.Println("\nCreating workforce pool: ", s.APIParams.PoolName)
+	poolFullName := fmt.Sprintf("locations/global/workforcePools/%s", s.APIParams.PoolName)
+	createPool := workforceService.Create(
+		"locations/global",
+		&iam.WorkforcePool{
+			Name:        poolFullName,
+			DisplayName: s.APIParams.PoolName,
+			Description: "Workforce pool created by Teleport",
+			Parent:      fmt.Sprintf("organizations/%s", s.APIParams.OrganizationID),
+		})
+	createPool.WorkforcePoolId(s.APIParams.PoolName)
+	resp, err := createPool.Do()
+	if err != nil {
+		// TODO(sshah): parse through error type for better error handling
+		return trace.Wrap(err)
+	}
+	fmt.Println("Pool created.")
+	if !resp.Done {
+		// 2 minutes timeout is semi-random decision, chosen based on the fact that
+		// when creating workforce pool from the GCP web console, it mentions
+		// that the operation could take up to 2 minutes.
+		pollCtx, cancel := context.WithTimeout(ctx, 120*time.Second)
+		defer cancel()
+
+		if err := waitForPoolStatus(pollCtx, workforceService, poolFullName, s.APIParams.PoolName); err != nil {
+			return trace.Wrap(err)
+		}
+		fmt.Printf("Pool %q is ready for use.\n", s.APIParams.PoolName)
+	}
+
+	fmt.Println("\nCreating workforce pool provider: ", s.APIParams.PoolProviderName)
+	metadata, err := fetchIdPMetadata(s.APIParams.SAMLIdPMetadataURL, s.HTTPClient)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	provider := &iam.WorkforcePoolProvider{
+		Description: "Workforce pool provider created by Teleport",
+		Name:        fmt.Sprintf("locations/global/workforcePools/%s/providers/%s", s.APIParams.PoolName, s.APIParams.PoolProviderName),
+		DisplayName: s.APIParams.PoolProviderName,
+		Saml: &iam.GoogleIamAdminV1WorkforcePoolProviderSaml{
+			IdpMetadataXml: metadata,
+		},
+		AttributeMapping: map[string]string{
+			"google.subject": "assertion.subject",
+			"google.groups":  "assertion.attributes.roles",
+		},
```

Comment:
Can we extract this and similar values to consts / package vars?

Reply:
I am leaving this as is for now. I am thinking if we should expose attribute mapping as config param so definitely need to work on this in a followup.